scylla/src/transport/large_batch_statements_test.rs

```diff
@@ -0,0 +1,67 @@
+use assert_matches::assert_matches;
+
+use scylla_cql::errors::{BadQuery, QueryError};
+
+use crate::batch::BatchType;
+use crate::query::Query;
+use crate::{
+    batch::Batch,
+    test_utils::{create_new_session_builder, unique_keyspace_name},
+    QueryResult, Session,
+};
+
+#[tokio::test]
+async fn test_large_batch_statements() {
+    let mut session = create_new_session_builder().build().await.unwrap();
+
+    let ks = unique_keyspace_name();
+    session = create_test_session(session, &ks).await;
+
+    let max_queries = u16::MAX as usize;
+    let batch_insert_result = write_batch(&session, max_queries, &ks).await;
+
+    batch_insert_result.unwrap();
+
+    let too_many_queries = u16::MAX as usize + 1;
+    let batch_insert_result = write_batch(&session, too_many_queries, &ks).await;
+    assert_matches!(
+        batch_insert_result.unwrap_err(),
+        QueryError::BadQuery(BadQuery::TooManyQueriesInBatchStatement(_too_many_queries)) if _too_many_queries == too_many_queries
+    )
+}
+
+async fn create_test_session(session: Session, ks: &String) -> Session {
+    session
+        .query(
+            format!("CREATE KEYSPACE IF NOT EXISTS {} WITH REPLICATION = {{ 'class' : 'NetworkTopologyStrategy', 'replication_factor' : 1 }}",ks),
+            &[],
+        )
+        .await.unwrap();
+    session
+        .query(format!("DROP TABLE IF EXISTS {}.pairs;", ks), &[])
+        .await
+        .unwrap();
+    session
+        .query(
+            format!("CREATE TABLE IF NOT EXISTS {}.pairs (dummy int, k blob, v blob, primary key (dummy, k))", ks),
+            &[],
+        )
+        .await.unwrap();
+    session
+}
+
+async fn write_batch(session: &Session, n: usize, ks: &String) -> Result<QueryResult, QueryError> {
+    let mut batch_query = Batch::new(BatchType::Unlogged);
+    let mut batch_values = Vec::new();
+    for i in 0..n {
+        let mut key = vec![0];
+        key.extend(i.to_be_bytes().as_slice());
+        let value = key.clone();
+        let query = format!("INSERT INTO {}.pairs (dummy, k, v) VALUES (0, ?, ?)", ks);
+        let values = vec![key, value];
+        batch_values.push(values);
+        let query = Query::new(query);
```

Comment:
How about using prepared statements instead? Prepare the statement once before creating the batch and then use it in `append_statement`: `batch_query.append_statement(prepared.clone())`. This should reduce the work needed by DB to process the batch.